Detectors/TPC/simulation/src/Detector.cxx

```diff
@@ -250,12 +253,41 @@ Bool_t  Detector::ProcessHits(FairVolume* vol)
     return kFALSE;
   
   // ADD HIT
-  TLorentzVector position;
+  static TLorentzVector position;
   refMC->TrackPosition(position);
   float time    = refMC->TrackTime() * 1.0e09;
   int trackID = refMC->GetStack()->GetCurrentTrackNumber();
   int detID   = vol->getMCid();
+
+#ifdef NEWHIT
+  static int oldTrackId = trackID;
+  static int oldDetId = detID;
+  static int groupCounter = 0;
+
+  //  a new group is starting -> put it into the container
+  static LinkableHitGroup *currentgroup = nullptr;
+  if (groupCounter == 0) {
+    TClonesArray& clref = *mHitGroupCollection;
+
+    // push-back in place
+    Int_t size = clref.GetEntriesFast();
+    currentgroup = new(clref[size]) LinkableHitGroup(trackID);
+
+    // set the MC truth link for this group
+    currentgroup->SetLink(FairLink(-1, mEventNr, mMCTrackBranchId, trackID)); 
```

Comment:
Is the trackID required in LinkableHitGroup? It is also set in the FairLink. So it looks like duplication of information...